client/src/app/pages/archetypes/archetypes-page.tsx

```diff
@@ -270,8 +277,33 @@ const Archetypes: React.FC = () => {
         </ConditionalRender>
       </PageSection>
 
-      {/* TODO: Add create/edit modal */}
+      {/* Create modal */}
+      <Modal
+        title={t("dialog.title.newArchetype")}
+        variant="medium"
+        isOpen={openCreateArchetype}
+        onClose={() => setOpenCreateArchetype(false)}
+      >
+        <ArchetypeForm onClose={() => setOpenCreateArchetype(false)} />
+      </Modal>
+
+      {/* Edit modal */}
+      <Modal
+        title={t("dialog.title.updateArchetype")}
+        variant="medium"
+        isOpen={!!archetypeToEdit}
+        onClose={() => setArchetypeToEdit(undefined)}
```

Comment:
Wondering if we should be using null here. This goes against my understanding of undefined usage vs null within our project. I was thinking we should use null when intentionally setting something as empty/not available & reserve undefined for something unintentionally empty / loading state from restful request or otherwise unexpected cases.

I think its acceptable and perfectly valid to use undefined in this case - just want to be clear on what we decide to use across the project for cases like this.

Reply:
I can swap it to `null`, it is just adding and extra `| null` to the useState type param and a few adjustments.